src/wp-includes/widgets.php

```diff
@@ -1350,6 +1350,13 @@ function retrieve_widgets( $theme_changed = false ) {
 	$sidebars_widgets = _wp_remove_unregistered_widgets( $sidebars_widgets, $registered_widgets_ids );
 	$sidebars_widgets = wp_map_sidebars_widgets( $sidebars_widgets );
 
+	// Replace null values inside the array with an empty array.
+	foreach ( $sidebars_widgets as $key => $value ) {
+		if ( null === $value ) {
```

Comment:
What if it was set to something other than an array or null, like a string, boolean, integer, or object? Wouldn't this be more appropriate:

```suggestion
		if ( ! is_array( $value ) ) {
```

Reply:
moving forward with this approach